python/paddle/fluid/layers/nn.py

```diff
@@ -2966,6 +2966,18 @@ def layer_norm(input,
     helper = LayerHelper('layer_norm', **locals())
     dtype = helper.input_dtype()
 
+    # cast to fp32 if using fp16
+    if dtype == core.VarDesc.VarType.FP16:
```

Comment:
Should put this in model configurations? Explicit should be better?

Reply:
Agreed and made the changes